src/app/shared/utils/ck-upload-adapter.ts

```diff
@@ -0,0 +1,83 @@
+import { DocumentService } from 'app/services/document.service';
+
+// This utility function is a template directly from CK,
+// modified to use our existing DocumentService.
+// https://ckeditor.com/docs/ckeditor5/latest/framework/guides/deep-dive/upload-adapter.html
+
+export class CkUploadAdapter {
+  private imageURL: string;
+  private imageUploadData: FormData;
+
+  constructor(
+    private loader,
+    private projectId: string,
+    private documentService: DocumentService,
+    private pathAPI: string
+  ) {
+    this.imageUploadData = new FormData();
+  }
+
+  /**
+   * Required function for the CKUploadAdapter.
+   * Uploads the file using the DocumentService,
+   * Then returns the image URL wrapped in a Promise.
+   *
+   * @returns {Promise}
+   */
+  upload(): Promise<any> {
```

Comment:
```suggestion
  upload(): Promise<{ default: string }|HTTPErrorResponse> {
```
I'm not 100% on the error response type...

Reply:
This change breaks the file, and prevents it from compiling locally. Not implementing.

Reply:
Yeah I didn't think this one would be so easy. Probably some narrowing that needs to be done here. If you're curious, see: https://www.typescriptlang.org/docs/handbook/2/narrowing.html

But yes, I'm alright with not implementing it for now.